### ISSUE TITLE
Multiple ShinyProxy Servers estart

### ISSUE THREAD

**Issue by kramerrs** [openanalytics/shinyproxy#504]
Accidentally posted comment while editing see below for comment.

**Comment by kramerrs** [comment]
The issue is that when we have multiple ShinyProxies running on Docker Swarm and have enabled reclaiming containers ShinyProxy grabs containers from other processes.

An additional yet related issue is that running multiple URL as endpoints on the same ShinyProxy are not supported, because landing pages are not configurable per app instance.

**Comment by LEDfan** [comment]
Hi

> The issue is that when we have multiple ShinyProxies running on Docker Swarm and have enabled reclaiming containers ShinyProxy grabs containers from other processes.

If I understand correctly, you are using [App Recovery](https://shinyproxy.io/documentation/app-recovery/)? Currently it's not possible to use app recovery together with pre-initialized containers. The reason is that we cannot persist the user apps using the labels of the docker containers (since they are not really related). However, it's possible to use Redis for persisting the sessions, allowing you to restart ShinyProxy, see: https://shinyproxy.io/documentation/configuration/#session-and-app-persistence

I see we did not yet mention in the docs that app recovery isn't supported, therefore I just updated the docs. In addition the next version of ShinyProxy will produce an error when app recovery is enabled together with a pre-initialized container.
